### PR TITLE
ENH: Add partial module data flag

### PR DESCRIPTION
**Pull request by nawtrey**
~* Add module `plot_flag` with functions for generating a
generic warning flag with configurable warning message~

* Add warning flags to summary report module table
for modules marked as having partial module data

* `test_module_table()` changes:

  - Add tests for logs `partial_data_stdio.darshan` and
  `partial_data_dxt.darshan` for checking cases with
  partial module data

  - Adjust actual and expected module dataframes
  to reflect changes in the module table

* Fix gh-498

### CODE REVIEW

**Comment by nawtrey** [comment]
Here are example reports for the 2 log files with partial module data: [report_examples.tar.gz](https://github.com/darshan-hpc/darshan/files/7952781/report_examples.tar.gz)

The current message is `"Ran out of memory or record limit reached!"`, hopefully @carns can comment on whether that's the ideal message for this use case.

File: darshan-util/pydarshan/darshan/cli/summary.py [hunk 1]
```diff
@@ -244,14 +245,18 @@ def get_module_table(self):
...
             val = f"{mod_buf_size:.2f} KiB"
+            flag = ""
             if self.report.modules[mod]["partial_flag"]:
-                val += " (partial data)"
-            module_dict[key] = val
+                msg = "Ran out of memory or record limit reached!"
+                fig = plot_flag.plot_flag(warn_msg=msg, font_size=20)
+                encoded = ReportFigure.get_encoded_fig(mpl_fig=fig)
+                flag = f"<img src=data:image/png;base64,{encoded} alt='Warning flag' width=110>"
+            module_dict[key] = [val, flag]
```
[the remark this answers is not in the document]
Comment (nawtrey) [reply darshan-util/pydarshan/darshan/cli/summary.py:254]: I wasn't aware of that unicode symbol, but this is embarrassingly simple in comparison to my approach. I don't really see a reason not to go with the unicode version, it's not like the custom warning sign is any more noticeable. We can probably add some inline CSS to wrap the text if we want, but I think it's fine either way.

[the remark this answers is not in the document]
Comment (nawtrey) [reply darshan-util/pydarshan/darshan/cli/summary.py:251]: Okay, I switched this to use the much simpler unicode warning symbol.

**Comment by tylerjereddy** [comment]
There's still at least one stale comment about image tags that doesn't make sense anymore:

`# check that number of img tags matches expected partial flag count`

I'll let you clean that up later...

Apart from that, I checked a few logs and it seems "ok." The old perl report had less granularity with the partial warnings, and seems to fail to warn for the partial dxt log entirely.

**Pull request merged**